### DISCUSSION TITLE
How to fix SpreadOperator on Room DI with Dagger

### DISCUSSION THREAD

**Discussion by alexis-depalle** [detekt/detekt#4527] [Q&A] [answered]
Hello,

I just activate the performance rules in my project and i got some errors due to SpreadOperator. I fix some errors but i'm facing a problem with room DI with dagger : 

fun database(context: Context): Database =
        Room.databaseBuilder(context, Database::class.java, context.packageName)
            .fallbackToDestructiveMigrationFrom(1)
            .addMigrations(*Database.migrations)
            .build()
            
Database.migrations's type is : `Array<Migration>`
            
Can someone explains me how to fix this issues please ?

Thanks a lot

**Comment by schalkms** [comment, marked as answer]
Did you try to use the "@Suppress" annotation accordingly for the `SpreadOperator`?

**Reply by alexis-depalle** [reply]
Yes, the suppress annotation works but it's not a fix, it's just a bypass no ?

**Reply by schalkms** [reply]
I guess it's fine to suppress this issue here.
You could also try to suppress the function with a **function suppressor** depending on what you favor.
https://detekt.github.io/detekt/suppressors.html#function-suppressor

**Reply by TWiStErRob** [reply]
Reference: https://github.com/detekt/detekt/issues/5510

**Reply by BraisGabin** [reply]
I'm still curious if this is still a performance issue on current kotlin versions.

Anyway, that code that you show will be run only once. So, the performance hit is negligible. This is only an issue if it is inside a loop.

~The problem is that we can't know if a function with this pattern is called inside a loop or not, so, for that reason we always flag it.~ Maybe we could do something to know if that code is inside a loop. Right now detekt can not handle such an advanced check but it would be nice to have something like that.
